scripts/consensus_merge.py

```diff
@@ -0,0 +1,587 @@
+            elif self.type == 'tumor':
+                # Strelka2 SGT field may be e.g. 'ref->het' or 'GG->GC'
+                strelka_sgt = self.record.info['SGT'].split("->")[-1]
+                try:
+                    tumor_gt = Sample.name_to_gt(strelka_sgt)
+                except ValueError:
+                    if len(set(strelka_sgt)) == 2:
+                        tumor_gt = '0/1'
+                    elif len(set(strelka_sgt)) == 1:
+                        if strelka_sgt[0] == self.record.ref:
+                            tumor_gt = '0/0'
+                        else:
+                            tumor_gt = '1/1'
+                    else:
+                        raise IOError('Unhandled Strelka2 SGT value %s' % record.info['SGT'])
+                return tumor_gt
+
+        elif self.caller in ('mutect2', 'vardict', 'lancet'):
+            return stringify(self.vcf_sample['GT'], '/')
+
+        else:
+            raise IOError('No rule to get GT for caller %s' % self.caller)
+
+    @functools.cached_property
+    def AD(self):
+        """ Instructions for AD FORMAT attribute """
+        if self.caller == 'strelka2':
+            """ From strelka2 docs:
+                SNPs:
+                    refCounts = Sum values FORMAT column $REF + “U” 
+                        (e.g. if REF="A" then use the value in FORMAT/AU)
+                    altCounts = Sum values FORMAT column $ALT + “U” 
+                        (e.g. if ALT="T" then use the value in FORMAT/TU)
+                Indels:
+                    tier1RefCounts = Sum values from FORMAT/TAR
+                    tier1AltCounts = Sume values from FORMAT/TIR
+            """
+            # only indels should have TIR field
+            if 'TIR' in self.vcf_sample:
+                ref_ct = sum(self.vcf_sample['TAR'])
+                alt_ct = sum(self.vcf_sample['TIR'])
+            else: # SNP case
+                snp_ref_key = self.record.ref[0] + "U"
+                snp_alt_key = self.record.alts[0][0] + "U"
+                ref_ct = sum(self.vcf_sample[snp_ref_key])
+                alt_ct = sum(self.vcf_sample[snp_alt_key])
+            
+            return (ref_ct, alt_ct)
+
+        elif self.caller in ('mutect2', 'vardict', 'lancet'):
+            return self.vcf_sample['AD']
+
+        else:
+            raise IOError('No rule to get AD for caller %s' % self.caller)
+
+    @functools.cached_property
+    def AF(self):
+        """ Instructions for AF FORMAT attribute """
+
+        if self.caller == 'mutect2':
+            return self.vcf_sample['AF'][0]
+        elif self.caller == 'vardict':
+            return self.vcf_sample['AF']
+        elif self.caller == 'lancet':
+            return Sample.AF_from_AD(self.vcf_sample['AD'])
+        elif self.caller == 'strelka2':
+            return Sample.AF_from_AD(self.AD)
+        else:
+            raise IOError('No rule to get AF for caller %s' % self.caller)
+
+    @functools.cached_property
```

Comment:
Interested in learning about some of the decorators that you've used

Reply:
This one just calculates the attribute on the fly the first time it's referenced and then remembers the calculated value thereafter. It had a different name prior to Python 3.8, which is probably the cause of the error you experienced running it. I will provide environment information once we settle on the VCF format and this becomes a 'real' PR.

Reply:
Ah, ok, that's fine, thanks!